libs/openchallenges/challenge-search/src/lib/challenge-search.component.ts

```diff
@@ -167,17 +171,28 @@ export class ChallengeSearchComponent
       minStartDate: this.selectedYear?.start || undefined,
       maxStartDate: this.selectedYear?.end || undefined,
     } as ChallengeSearchQuery;
-    this.query.next(defaultQuery);
+
+    this.activatedRoute.queryParams.subscribe((params) => {
+      // only support querying by searchTerm from query url for now
+      if (params['searchTerms']) {
+        this.searchTermValue = params['searchTerms'];
+        defaultQuery['searchTerms'] = params['searchTerms'];
+      }
+      this.query.next(defaultQuery);
+    });
   }
 
   ngAfterContentInit(): void {
     this.searchTerms
-      .pipe(debounceTime(400), distinctUntilChanged(), takeUntil(this.destroy))
+      .pipe(
+        skip(1),
+        debounceTime(400),
+        distinctUntilChanged(),
+        takeUntil(this.destroy)
+      )
       .subscribe((search) => {
```

Comment:
Could we rename the parameter `searchTerms` and then later do `{ searchTerms }` instead of `{ searchTerms: search}`?